webapp/src/DOMJudgeBundle/Controller/API/JudgehostController.php

```diff
@@ -689,185 +701,91 @@ public function addJudgingRunAction(
-        $runs = $this->entityManager->createQueryBuilder()
-            ->from('DOMJudgeBundle:JudgingRun', 'r')
-            ->join('r.testcase', 't')
-            ->select('r')
-            ->andWhere('r.judgingid = :judgingid')
-            ->orderBy('t.rank')
-            ->setParameter(':judgingid', $judgingId)
-            ->getQuery()
-            ->getResult();
-
-        $numTestCases = $this->entityManager->createQueryBuilder()
-            ->from('DOMJudgeBundle:Testcase', 't')
-            ->select('COUNT(t.testcaseid)')
-            ->where('t.probid = :probid')
-            ->setParameter(':probid', $testCase->getProbid())
-            ->getQuery()
-            ->getSingleScalarResult();
-
-        $allRuns = array_pad($runs, (int)$numTestCases, null);
-
-        $oldResult = $judging->getResult();
-
-        if (($result = $this->submissionService->getFinalResult($allRuns, $resultsPrio)) !== null) {
-            // Lookup global lazy evaluation of results setting and possible problem specific override.
-            $lazyEval    = $this->DOMJudgeService->dbconfig_get('lazy_eval_results', true);
-            $problemLazy = $judging->getSubmission()->getContestProblem()->getLazyEvalResults();
-            if (isset($problemLazy)) {
-                $lazyEval = $problemLazy;
-            }
-
-            $judging->setResult($result);
-            if (count($runs) == $numTestCases || $lazyEval) {
-                // NOTE: setting endtime here determines in testcases_GET
-                // whether a next testcase will be handed out.
-                $judging->setEndtime(Utils::now());
-            }
-            $this->entityManager->flush();
-
-            // Only update if the current result is different from what we had before.
-            // This should only happen when the old result was NULL.
-            if ($oldResult !== $result) {
-                if ($oldResult !== null) {
-                    throw new \BadMethodCallException('internal bug: the evaluated result changed during judging');
-                }
-
-                $submission = $judging->getSubmission();
-                $contest    = $submission->getContest();
-                $team       = $submission->getTeam();
-                $problem    = $submission->getProblem();
-                $this->scoreboardService->calculateScoreRow($contest, $team, $problem);
-
-                // We call alert here before possible validation. Note that this means that these
-                // alert messages should be treated as confidential information.
-                $this->DOMJudgeService->alert($result === 'correct' ? 'accept' : 'reject',
-                                              sprintf("submission %s, judging %s: %s",
-                                                      $submission->getSubmitid(),
-                                                      $judging->getJudgingid(), $result));
+        // FIXME: why don't we do a similar check as three lines above?
```

Comment:
Assuming you mean checking whether the judging exists: we definitely should

Reply:
Done.